### PR TITLE
Add maximum height above ground (HAGL) check to all Navigator modes

### PR DESCRIPTION
**Pull request by bresch**
### Solved Problem
The HAGL navigator failsafe I added in https://github.com/PX4/PX4-Autopilot/pull/23501 should be used in all navigator modes.

### Solution
Centralize the check instead of adding it to each mode separately.

### Changelog Entry
For release notes:
```
Add maximum height above ground (HAGL) check to all Navigator modes
New parameter: -
Documentation: -
```

### Test coverage
- SITL tests

### CODE REVIEW

**Review by sfuhrer — CHANGES_REQUESTED** [review]
Agree that we should aim for a generic solution and not have them custom for RTL/Takeoff. Some comments from testing in SITL:
- the reporting of the HAGL failsafe is a bit rough. E.g. it says it will switch to Hold when it already is in Hold, and it doesn't apply the timeout
image
- doesn't seem to work with MC Orbit
- RTL wih OF only: why does it 3 times switch into RTL before then going to Land?
image

File: src/modules/navigator/mission_block.cpp [hunk 1]
```diff
@@ -1042,3 +1042,29 @@ void MissionBlock::updateAltToAvoidTerrainCollisionAndRepublishTriplet(mission_i
...
+	updateMaxHaglFailsafe();
+}
+
+void MissionBlock::updateMaxHaglFailsafe()
+{
+	const float target_alt = _navigator->get_position_setpoint_triplet()->current.alt;
+
+	if (_navigator->get_global_position()->terrain_alt_valid
+	    && ((target_alt - _navigator->get_global_position()->terrain_alt) > _navigator->get_local_position()->hagl_max)) {
+		// Handle case where the takeoff altidude is above the maximum HAGL
```
[the remark this answers is not in the document]
Comment (bresch) [reply src/modules/navigator/mission_block.cpp:1057]: Hmm, that's a good example why comments are bad

File: src/modules/navigator/mission_block.cpp [hunk 2]
```diff
@@ -1042,3 +1042,29 @@ void MissionBlock::updateAltToAvoidTerrainCollisionAndRepublishTriplet(mission_i
...
+		mavlink_log_info(_navigator->get_mavlink_log_pub(), "Target altitude higher than max HAGL\t");
+		events::send(events::ID("navigator_takeoff_fail_max_hagl"), events::Log::Error, "Target altitude higher than max HAGL");
```
[the remark this answers is not in the document]
Comment (bresch) [reply src/modules/navigator/mission_block.cpp:1059]: thanks

**Review by sfuhrer — APPROVED** [review]
Looks good to me now, the extension to not have the delay when switching to Hold seems correct. For my other comments:

>doesn't seem to work with MC Orbit

Okay that is expected and not easy to change, as the Orbit doesn't use the triplet at all

> RTL wih OF only: why does it 3 times switch into RTL before then going to Land?

That was actually the GCS sending the command 3 times.

**Pull request merged**